File: backend/PayProModulith/src/main/java/com/example/paypromodulith/authentication/domain/model/AdminDto.java
package com.example.paypromodulith.authentication.domain.model;

import java.io.Serializable;
import java.time.Instant;
import java.util.LinkedHashSet;
import java.util.Set;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/** DTO for {@link com.example.paypromodulith.entity.Admin} */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AdminDto implements Serializable {
  private UUID id;
  private Instant createdAt;
  private String name;
  private String email;
  private String password;
  private RoleDto role;
  private Set<AdminDepartmentDto> adminDepartments = new LinkedHashSet<>();
  private Set<AdminRowDto> adminRows = new LinkedHashSet<>();
  private SpaceDto space;
}

File: frontend/web/PayProWeb/src/app/model/AdminDto.ts
import {AdminRowDto} from './AdminRowDto';

export interface AdminDto {
	id?: string;
	name: string;
	email: string;
	password: string;
  phone? : string;
  address? : string;
	role: RoleDto;
	adminDepartments?: AdminDtoAdminDepartments[];
	adminRows?: AdminRowDto[];
	space: AdminDtoSpace;

}
export interface RoleDto {
	id: string;
	name: string;
}
export interface AdminDtoAdminDepartmentsDepartmentOrganisation {
	id: string;
	name: string;
}
export interface AdminDtoAdminDepartmentsDepartment {
	id: string;
	createdAt: string;
	name: string;
	organisation: AdminDtoAdminDepartmentsDepartmentOrganisation;
}
export interface AdminDtoAdminDepartments {
	id: number;
	createdAt: string;
	department: AdminDtoAdminDepartmentsDepartment;
}
export interface AdminDtoAdminRowsOrganisation {
	id: string;
	name: string;
}
export interface AdminRows {
	id: number;
	organisation: AdminDtoAdminRowsOrganisation;
}
export interface OrganisationDto {
	id: string;
	name: string;
}
export interface AdminDtoSpace {
	id: string;
	organisationDtos: OrganisationDto[];
}
